ReactNativeClient/lib/joplin-renderer/utils.js

```diff
@@ -138,7 +138,7 @@ utils.imageReplacement = function(ResourceModel, src, resources, resourceBaseUrl
 	}
 
 	const mime = resource.mime ? resource.mime.toLowerCase() : '';
-	if (ResourceModel.isSupportedImageMimeType(mime)) {
+	if (ResourceModel.isSupportedImageMimeType(mime) || ResourceModel.isSupportedAudioMimeType(mime)) {
```

Comment:
In line with the comment above, this function (`imageReplacement`) should be generalized to `resourceReplacement` and this line should be changed. In the future there may be more mimeTypes added so it would be nice to have a more flexible check here. 

Why did you have multiple function calls here instead of having one long array of supported mimeTypes?

Reply:
For example isSupportedImageMimeType could return a resource type (image, audio, video, etc.) which would then be included in the return of this function. Back in the rule you would then perform the check `if (r && r.type === "audio")` this way there would be solid support for other mimeTypes. As it stands now this code is limited to just audio (and image)

Reply:
The reason I have two separate function calls is because currently when you attach a file, the code uses isSupportedImageMimeType to add an exclamation mark before the resource if it is an image. So if I make the changes you are suggesting here I think that would be good. Also the logic for adding the exclamation mark would be changed to use the resource type that is returned from isSupportedMimeType.